src/components/inputs/localized-money-input/README.md

```diff
@@ -0,0 +1,173 @@
+If you want to forbid `highPrecision`, then the form's validation needs to add an error when it sees a `highPrecision` price. See example below.
+
+Here are examples of `centPrecision` and `highPrecision` prices.
+
+```js
+// 42.00 €
+{
+  "type": "centPrecision",
+  "currencyCode": "EUR",
+  "centAmount": 4200,
+  "fractionDigits": 2
+}
+```
+
+```js
+// 0.0123456 €
+{
+ "type": "highPrecision",
+ "currencyCode": "EUR",
+ "centAmount": 1,
+ "preciseAmount": 123456,
+ "fractionDigits": 7
+}
+```
+
+#### `LocalizedMoneyInput.parseMoneyValue`
+
+The `parseMoneyValue` function will turn a [`MoneyValue`](https://docs.commercetools.com/http-api-types#money) into a value the LocalizedMoneyInput component can handle `({ amount, currencyCode })`.
+
+##### `isEmpty(localizedField)`
+
+Returns `true` when all values in a localized field are empty.
+
+```js
+LocalizedMoneyInput.isEmpty({});
+// -> true
+```
+
+```js
+LocalizedMoneyInput.isEmpty({ USD: '', EUR: '  ' });
+// -> true
+```
+
+```js
+LocalizedMoneyInput.isEmpty({ USD: '12.43', EUR: '' });
+// -> false
+```
+
+##### `LocalizedMoneyInput.convertToMoneyValues`
+
+The `convertToMoneyValues` function will turn a LocalizedMoneyInput value into array of [`MoneyValue`](https://docs.commercetools.com/http-api-types#money) the API can handle. It automatically converts to `centPrecision` or `highPrecision` types when the number of supplied fraction digits exceeds the number of fraction digits used by the currency.
+If you want to forbid `highPrecision`, then the form's validation needs to add an error when it sees a `highPrecision` price. See example below.
+
+Here are examples of `centPrecision` and `highPrecision` prices.
+
+```js
+// 42.00 €
+[
+  {
+    type: 'centPrecision',
+    currencyCode: 'EUR',
+    centAmount: 4200,
+    fractionDigits: 2,
+  },
+];
+```
+
+```js
+// 0.0123456 €
+[
+  {
+    type: 'highPrecision',
+    currencyCode: 'EUR',
+    centAmount: 1,
+    preciseAmount: 123456,
+    fractionDigits: 7,
+  },
+];
+```
+
+##### `LocalizedMoneyInput.parseMoneyValues`
+
+The `parseMoneyValues` function will turn a [`MoneyValue`](https://docs.commercetools.com/http-api-types#money) into a value the LocalizedMoneyInput component can handle `({ amount, currencyCode })`.
+
+##### `LocalizedMoneyInput.isEmpty`
+
+The `isEmpty` function will return `true` for the crrency when the passed `LocalizedMoneyInput` value for this currency is empty (either has no currency or no amount, or does not exist at all).
+
+```js
+LocalizedMoneyInput.isEmpty([{ amount: '', currencyCode: 'EUR' }]); // -> { EUR: true }
+
+LocalizedMoneyInput.isEmpty({ amount: '5', currencyCode: 'EUR' }); // -> { EUR: false }
```

Comment:
@islam3zzat I know we spoke about this before. However this isn't quite what I had expected (or maybe I'm remembering it wrong). Here's my current take for how things are supposed to work. I'm happy to chat to you again to align on the approach.

Generally, `LocalizedMoneyInput` should only operate on the `value` it expects (instead of on an array with `amount` and `currencyCode`).

I would also make `LocalizedMoneyInput.isEmpty` the only way to check whether the value is empty. So no `isAllEmpty` and no `hasEmptyValue`.

---

To me, `LocalizedMoneyInput.isEmpty` should be designed to work with the `value` passed to `LocalizedMoneyInput` which is an object (not an array as shown some of the examples). So `LocalizedMoneyInput.isEmpty` should accept an object instead of an array.

We should be able to do `LocalizedMoneyInput.isEmpty({ EUR: "12.00", USD: "14.23", CAD: "" })`, which would return `false` as at least one non-empty currency-amount pair is provided. That way it works simliar to `TextInput.isEmpty` which also returns `false` when at least one translation is provided.

This means we can remove `isAllEmpty` and `hasEmptyValue`.

---

There might be a case where users need to provide a value for every currency. We can accept

```js
LocalizedMoneyInput.isEmpty(
  { EUR: "12.00", USD: "14.23", CAD: "" },
  { requireAllCurrenciesToBeFilled: true }
)
```

for such cases. It might also be useful for consumers to be able to determine the currencies with empty values. We can make this easier by providing a separate function which returns this information (either as an array or as an object; not sure which API is better):

```js
// as an array
LocalizedMoneyInput.getEmptyCurrencies({ EUR: "12.00", USD: "14.23", CAD: "" })
// -> ["CAD"]

// as an object
LocalizedMoneyInput.getEmptyCurrencies({ EUR: "12.00", USD: "14.23", CAD: "" })
// -> { EUR: false, USD: false, CAD: true }
```

I slightly prefer the first version as it's easier to do `emptyCurrencies.includes("CAD").


However I wouldn't build this functionality until we actually need it. So let's leave it out for now?

Reply:
You are right, the statics here should operate on the value it expects.